nwapp/utils/notifier.js

```diff
@@ -28,14 +36,85 @@ function playNotificationSound() {
   audio = null;
 }
 
+
+// Cache lookup
+var urlFilePathMap = new Map();
+function getFileAtUrl(url) {
```

Comment:
If you return a file url (`file:///my/image/whatever.png`), then your code that uses this function can be a little bit simpler as it will only have to deal with urls.